Create ConfirmedPlaces for existing ConfirmedTrips

JGreenlee opened the pull request:
Fixes https://github.com/e-mission/e-mission-docs/issues/865

Before confirmed_place was introduced, we created confirmed_trips without a confirmed_place. So, these existing entries are missing the confirmed_place ID for the place they should be linked to.
For those trips, we will generate a confirmed_place just-in-time and add its ID to the trip.

Once all trips have a confirmed_place, we can remove this code.

JGreenlee committed:
fix missing confirmed_places on existing confirmed_trips

JGreenlee committed:
insert confirmed_place_entry to ts

shankari commented:
> Once all trips have a confirmed_place, we can remove this code.

And all trips should have a confirmed place after the first successful run of the pipeline that includes a `CREATE_COMPOSITE_TRIP` step

shankari commented on emission/analysis/userinput/matcher.py:
```diff
@@ -87,6 +87,17 @@ def create_composite_objects(user_id):
+        # Once every trip has a confirmed_place, we can remove this code
+        if "confirmed_place" not in ct["data"]:
+            cleaned_place = esda.get_entry(esda.CLEANED_PLACE_KEY, ct["data"]["end_place"])
+            confirmed_place_entry = create_confirmed_place_entry(ts, cleaned_place)
+            ts.insert(confirmed_place_entry)
+            ct["data"]["confirmed_place"] = confirmed_place_entry["_id"]
```
@JGreenlee how is the `confirmed_place_entry['_id')` filled in?
`Entry.createEntry` fills it in, but you are simply using `new Entry()`
I assumed that you would get the id back from the insert call and use it

Can you verify that the `_id` field is actually non-empty?

```suggestion
            confirmed_place_entry = create_confirmed_place_entry(ts, cleaned_place)
            cpeid = ts.insert(confirmed_place_entry)
            ct["data"]["confirmed_place"] = cpeid
            logging.debug("Setting the confirmed_place key to the newly created id %s" % cpeid)
```

JGreenlee replied:
`_id` was empty before being added to the timeseries. That's how the first commit was.

The second commit calls ts.insert() first, after which `_id` does exist.

JGreenlee replied:
The suggestion looks cleaner though. Let's use that.

shankari reviewed (changes requested):
@JGreenlee Looks good overall - just want to confirm that one part is actually working before merging
Maybe confirm from the logs that the id is actually getting filled in correctly

JGreenlee committed:
receive confirmed place id from ts.insert()

applies suggestion from https://github.com/shankari/e-mission-server/pull/5#discussion_r1145281850

Co-authored-by: shankari <shankari@eecs.berkeley.edu>

shankari reviewed (approved):
LGTM! Let's deploy this and see if it works
